### ISSUE TITLE
crash during initialization when mapping module local rng and setting gobal rng by `seed-N-mt*` instead of `seed-N-mt`

### ISSUE THREAD

**Issue by Vonfry** [omnetpp/omnetpp#903]
**Platform**
OMNeT++ version: 
```
OMNeT++ Discrete Event Simulation  (C) 1992-2019 Andras Varga, OpenSim Ltd.
Version: 5.6.2, build: 200518-aa79d0918f, edition: Academic Public License -- NOT FOR COMMERCIAL USE
See the license for distribution terms and warranty disclaimer

Setting up Qtenv...

Build: omnetpp-5.6.2 200518-aa79d0918f
Compiler: GCC 10.3.0
Options: 64-bit DEBUG simtime_t=omnetpp::SimTime large-file-support=yes WITH_PARSIM WITH_MPI WITH_NETBUILDER WITH_QTENV XMLPARSER=libxml

End.
```
 OS: `NixOS 21.05 (Okapi)`

**Describe the bug**

Backtrace
```
==== backtrace (tid: 939640) ====
 0 0x00000000000282f5 ucs_debug_print_backtrace()  ???:0
 1 0x000000000003c3a0 killpg()  ???:0
 2 0x000000000003b13d omnetpp::common::PatternMatcher::matches()  /build/omnetpp-5.6.2/src/common/patternmatcher.cc:393
 3 0x000000000003b13d omnetpp::common::PatternMatcher::matches()  /build/omnetpp-5.6.2/src/common/patternmatcher.cc:394
 4 0x000000000005d10a omnetpp::envir::SectionBasedConfiguration::getMatchingPerObjectConfigKeys()  /build/omnetpp-5.6.2/src/envir/sectionbasedconfig.cc:1340
 5 0x0000000000059496 omnetpp::envir::SectionBasedConfiguration::getMatchingPerObjectConfigKeySuffixes()  /build/omnetpp-5.6.2/src/envir/sectionbasedconfig.cc:1354
 6 0x0000000000059496 std::vector<char const*, std::allocator<char const*> >::begin()  /nix/store/sjhz1j2d1ssn59f66kqp92xj9mpsww2d-gcc-10.3.0/include/c++/10.3.0/bits/stl_vector.h:812
 7 0x0000000000059496 omnetpp::envir::SectionBasedConfiguration::getMatchingPerObjectConfigKeySuffixes()  /build/omnetpp-5.6.2/src/envir/sectionbasedconfig.cc:1355
 8 0x0000000000055e42 omnetpp::envir::EnvirBase::setupRNGMapping()  /build/omnetpp-5.6.2/src/envir/envirbase.cc:1545
 9 0x0000000000055e42 std::vector<char const*, std::allocator<char const*> >::end()  /nix/store/sjhz1j2d1ssn59f66kqp92xj9mpsww2d-gcc-10.3.0/include/c++/10.3.0/bits/stl_vector.h:839
10 0x0000000000055e42 std::vector<char const*, std::allocator<char const*> >::empty()  /nix/store/sjhz1j2d1ssn59f66kqp92xj9mpsww2d-gcc-10.3.0/include/c++/10.3.0/bits/stl_vector.h:1008
11 0x0000000000055e42 omnetpp::envir::EnvirBase::setupRNGMapping()  /build/omnetpp-5.6.2/src/envir/envirbase.cc:1546
12 0x00000000001bcd92 omnetpp::cModuleType::create()  /build/omnetpp-5.6.2/src/sim/ccomponenttype.cc:328
13 0x00000000001bcd92 omnetpp::cModuleType::create()  /build/omnetpp-5.6.2/src/sim/ccomponenttype.cc:331
14 0x000000000023258d omnetpp::cSimulation::setupNetwork()  /build/omnetpp-5.6.2/src/sim/csimulation.cc:388
15 0x000000000004d587 omnetpp::envir::EnvirBase::setupNetwork()  /build/omnetpp-5.6.2/src/envir/envirbase.cc:771
16 0x00000000001720bc omnetpp::qtenv::Qtenv::setupNetwork()  /build/omnetpp-5.6.2/src/qtenv/qtenv.cc:1198
17 0x0000000000172aa0 omnetpp::qtenv::Qtenv::initialSetUpConfiguration()  /build/omnetpp-5.6.2/src/qtenv/qtenv.cc:1608
18 0x0000000000172aa0 omnetpp::qtenv::Qtenv::initialSetUpConfiguration()  /build/omnetpp-5.6.2/src/qtenv/qtenv.cc:1609
19 0x00000000003033e2 QTimer::timeout()  ???:0
20 0x00000000002f506f QObject::event()  ???:0
21 0x000000000017174f QApplicationPrivate::notify_helper()  ???:0
22 0x00000000002c83ca QCoreApplication::notifyInternal2()  ???:0
23 0x000000000031f8cb QTimerInfoList::activateTimers()  ???:0
24 0x000000000032018c QEventDispatcherGlib::flush()  ???:0
25 0x00000000000565ab g_main_context_dispatch()  ???:0
26 0x0000000000056858 g_main_context_iterate.constprop.0()  ???:0
27 0x000000000005690f g_main_context_iteration()  ???:0
28 0x0000000000320520 QEventDispatcherGlib::processEvents()  ???:0
29 0x00000000002c6d7b QEventLoop::exec()  ???:0
30 0x00000000002cf360 QCoreApplication::exec()  ???:0
31 0x0000000000170288 omnetpp::qtenv::Qtenv::doRun()  /build/omnetpp-5.6.2/src/qtenv/qtenv.cc:632
32 0x000000000004a356 omnetpp::envir::EnvirBase::run()  /build/omnetpp-5.6.2/src/envir/envirbase.cc:747
33 0x000000000004efec omnetpp::envir::EnvirBase::run()  /build/omnetpp-5.6.2/src/envir/envirbase.cc:358
34 0x000000000004649a omnetpp::envir::setupUserInterface()  /build/omnetpp-5.6.2/src/envir/startup.cc:259
35 0x0000000000047588 evMain()  /build/omnetpp-5.6.2/src/envir/evmain.cc:33
36 0x0000000000047588 evMain()  /build/omnetpp-5.6.2/src/envir/evmain.cc:32
37 0x0000000000027ded __libc_start_main()  ???:0
38 0x000000000048490a _start()  /build/glibc-2.32/csu/../sysdeps/x86_64/start.S:120
=================================
```

Extra infomation:
1. it is built with `-std=c++17`.
2. the sample `cqn` in omnetpp which contains module local rng mapping can work.

**To Reproduce**
This can be reproduced with [OPS](https://github.com/ComNets-Bremen/OPS) for me if setting something like `**.mobility.rng-0 = 6`.

This problem is happened when using `seed-N-mt* = M`.

For example:

The following is crash:
```
seed-1-mt* = 1
**.NODE.rng-0 = 1
```

The following is ok:
```
seed-1-mt = 1
**.NODE.rng-0 = 1
```

**Comment by avarga** [comment]
Thanks for reporting! I can reproduce it with samples/routing too, adding the two lines (with NODE=routing) into omnetpp.ini. I'll look into it in the next days.

**Comment by avarga** [comment]
Fixed in master (currently 6.0 devel) and in omnetpp-5.x. Latter will be released as 5.7.1 when enough bug fixes accumulate.
